src/pudl/analysis/plant_parts_eia.py

```diff
@@ -0,0 +1,1733 @@
+other plant-parts all in the same table without any clear labels. The master
+plant-part table is an attempt to create records corresponding to many
+different plant-parts in order to connect specific slices of EIA plants to
+other datasets.
+
+Because generators are often owned by multiple utilities, another dimention of
+the master unit list involves generating two records for each owner: one of the
+portion of the plant part they own and one for the plant part as a whole. The
+portion records are labeled in the ``ownership`` column as "owned" and the total
+records are labeled as "total".
+
+This module refers to "true granularies". Many plant parts we cobble together
+here in the master plant-part list refer to the same collection of
+infrastructure as other plant-part list records. For example, if we have a
+"plant_prime_mover" plant part record and a "plant_unit" plant part record
+which were both cobbled together from the same two generators. We want to be
+able to reduce the plant-part list to only unique collections of generators,
+so we label the first unique granularity as a true granularity and label the
+subsequent records as false granularities with the ``true_gran`` column. In
+order to choose which plant-part to keep in these instances, we assigned a
+:py:const:`PLANT_PARTS_ORDERED` and label whichever plant-part comes first as
+the unique granularity.
+
+**Recipe Book for the plant-part list**
+
+:py:const:`PLANT_PARTS` is the main recipe book for how each of the plant-parts
+need to be compiled. These plant-parts represent ways to group generators based
+on widely reported values in EIA. All of these are logical ways to group
+collections of generators - in most cases - but some groupings of generators
+are more prevelant or relevant than others for certain types of plants.
+
+The canonical example here is the ``plant_unit``. A unit is a collection of
+generators that operate together - most notably the combined-cycle natural gas
+plants. Combined-cycle units generally consist of a number of gas turbines
+which feed excess steam to a number of steam turbines.
+
+>>> df_gens = pd.DataFrame({
+...     'plant_id_eia': [1, 1, 1],
+...     'generator_id': ['a', 'b', 'c'],
+...     'unit_id_pudl': [1, 1, 1],
+...     'prime_mover_code': ['CT', 'CT', 'CA'],
+...     'capacity_mw': [50, 50, 100],
+... })
+>>> df_gens
+    plant_id_eia    generator_id    unit_id_pudl    prime_mover_code    capacity_mw
+0              1               a               1                  CT             50
+1              1               b               1                  CT             50
+2              1               c               1                  CA            100
+
+A good example of a plant-part that isn't really logical also comes from a
+combined-cycle unit. Grouping this example plant by the ``prime_mover_code``
+would generate two records that would basically never show up in FERC1.
+This stems from the inseparability of the generators.
```

Comment:
So the dataframe below is the result of grouping the dataframe from above by `prime_mover_code`? Why would this never show  up  in FERC1?

Reply:
Also, where did the `plant_part` column come from?